File: Day4/Puzzel2.py
counter=0
with open('input.txt') as inputFile:
    for line in inputFile:
        elf1 = line.split(',')[0]
        elf2 = line.split(',')[1]

        elf1begin = int(elf1.split('-')[0])
        elf1end = int(elf1.split('-')[1])

        elf2begin = int(elf2.split('-')[0])
        elf2end = int(elf2.split('-')[1])

        if( elf1begin >= elf2begin and elf1begin <= elf2end ):
            counter+=1
        elif( elf1end >= elf2begin and elf1end <= elf2end ):
            counter+=1
        elif( elf2begin >= elf1begin and elf2begin <= elf1end ):
            counter+=1
        elif( elf2end >= elf1begin and elf2end <= elf1end ):
            counter+=1

print( f"{counter=}" )

File: 2022/Day04/Puzzel2.cpp
#include <fstream>
#include <string>
#include <vector>
#include <iostream>


int intersects( const int &start1, const int &end1, const int &start2, const int &end2){
    return ( start1 >= start2 && start1 <= end2 ) ||
           ( end1 >= start2 && end1 <= end2 ) ||
           ( start2 >= start1 && start2 <= end1 ) ||
           ( end2 >= start1 && end2 <= end1 );
}

int main()
{
  std::vector<std::string> input = {};
  std::ifstream inputFile("input.txt");
  std::string temp;
  while( std::getline(inputFile, temp) )
  {
    input.push_back(temp);
  }
  int counter = 0;
  for( auto item: input )
  {
    std::string elf1 = item.substr( 0, item.find(',') );
    std::string elf2 = item.substr( item.find(',')+1, item.size()-item.find(',') );

    int beginElf1 = std::stoi( elf1.substr( 0, elf1.find('-') ) );
    int endElf1 = std::stoi( elf1.substr( elf1.find('-')+1, elf1.size()-elf1.find('-') ) );

    int beginElf2 = std::stoi( elf2.substr( 0, elf2.find('-') ) );
    int endElf2 = std::stoi( elf2.substr( elf2.find('-')+1,  elf2.size()-elf2.find('-') ) );
    
    counter += intersects( beginElf1, endElf1, beginElf2, endElf2 );
  }
  std::cout << counter << '\n';
}